test/unit/org/apache/cassandra/index/IndexStatusManagerTest.java

```diff
@@ -0,0 +1,393 @@
+                        .replicationStrategy(new NetworkTopologyStrategy("ks1", Map.of("DC", "5")))
+                        .indexStatus(Map.of(
+                                InetAddressAndPort.getByNameUnchecked("127.0.0.255"),
+                                Map.of(
+                                        "ks1.idx1", Index.Status.UNKNOWN,
+                                        "ks1.idx2", Index.Status.BUILD_SUCCEEDED,
+                                        "ks1.idx3", Index.Status.BUILD_SUCCEEDED
+                                ),
+                                InetAddressAndPort.getByNameUnchecked("127.0.0.254"),
+                                Map.of(
+                                        "ks1.idx1", Index.Status.BUILD_SUCCEEDED,
+                                        "ks1.idx2", Index.Status.BUILD_SUCCEEDED,
+                                        "ks1.idx3", Index.Status.UNKNOWN
+                                ),
+                                InetAddressAndPort.getByNameUnchecked("127.0.0.253"),
+                                Map.of(
+                                        "ks1.idx1", Index.Status.BUILD_SUCCEEDED,
+                                        "ks1.idx2", Index.Status.BUILD_SUCCEEDED,
+                                        "ks1.idx3", Index.Status.BUILD_SUCCEEDED
+                                ),
+                                InetAddressAndPort.getByNameUnchecked("127.0.0.252"),
+                                Map.of(
+                                        "ks1.idx1", Index.Status.BUILD_SUCCEEDED,
+                                        "ks1.idx2", Index.Status.BUILD_FAILED,
+                                        "ks1.idx3", Index.Status.BUILD_SUCCEEDED
+                                ),
+                                InetAddressAndPort.getByNameUnchecked("127.0.0.251"),
+                                Map.of(
+                                        "ks1.idx1", Index.Status.BUILD_SUCCEEDED,
+                                        "ks1.idx2", Index.Status.UNKNOWN,
+                                        "ks1.idx3", Index.Status.BUILD_SUCCEEDED
+                                )
+                        ))
+                        .minRequiredNodes(3)
+                        .expected(EndpointsForRange.of(
+                                full(InetAddressAndPort.getByNameUnchecked("127.0.0.255")),
+                                full(InetAddressAndPort.getByNameUnchecked("127.0.0.254")),
+                                full(InetAddressAndPort.getByNameUnchecked("127.0.0.253")),
+                                full(InetAddressAndPort.getByNameUnchecked("127.0.0.251"))
+                        ))
+                        .build()
+        );
+    }
+
+    // we don't have enough queryable endpoints: filterForQuery should throw an exception
+    @Test
+    public void filterForQuery_ThrowExceptionWhenNotEnoughQueryableEndpoints()
+    {
+        assertThatThrownBy(() ->
```

Comment:
I'd make this its own top-level test, called perhaps `shouldThrowWhenNotEnoughQueryableEndpoints()`